libiop/protocols/encoded/lincheck/basic_lincheck_aux.tcc

```diff
@@ -92,8 +93,30 @@ std::shared_ptr<std::vector<FieldT>> multi_lincheck_virtual_oracle<FieldT>::eval
     /* p_{alpha}^1 in [BCRSVW18], but now using the lagrange polynomial from 
      * [TODO: cite Succinct Aurora] instead of powers of alpha. */
     /* Compute p_alpha_prime. */
-    std::vector<FieldT> p_alpha_prime_over_codeword_domain = 
+    std::vector<FieldT> p_alpha_prime_over_codeword_domain;
+    
+
+    /* If |variable_domain| > |constraint_domain|, we multiply the Lagrange sampled 
+       polynomial by Z_{variable_domain}*Z_{constraint_domain}^-1 */
+    if (this->variable_domain_.num_elements() < this->constraint_domain_.num_elements()){
+        p_alpha_prime_over_codeword_domain = 
         this->p_alpha_.evaluations_over_field_subset(this->codeword_domain_);
+    }else{
+        /* inverses of the evaluations of constraint domain polynomial */
+        std::vector<FieldT> constraint_domain_vanishing_polynomial_inverses;
+        std::vector<FieldT> variable_domain_vanishing_polynomial_evaluations;
+        p_alpha_prime_over_codeword_domain = this->p_alpha_.evaluations_over_field_subset(this->codeword_domain_);
+
+        variable_domain_vanishing_polynomial_evaluations = this->variable_domain_vanishing_polynomial_
+                                                        .evaluations_over_field_subset(this->codeword_domain_);
+        constraint_domain_vanishing_polynomial_inverses = batch_inverse(this->constraint_domain_vanishing_polynomial_
+                                                        .evaluations_over_field_subset(this->codeword_domain_));
+
+        for (int i = 0; i < variable_domain_vanishing_polynomial_evaluations.size(); i++)
```

Comment:
Minor note, can you put the following line of code in a `{ }` block? Consistency with this sort of thing makes it easier to review the final code base. Hope to integrate linting soon, so there will be an automatic check for this sort of thing.